FWCore/Framework/src/Schedule.cc

```diff
@@ -243,7 +251,51 @@ namespace edm {
             std::string friendlyClassName = pset.getParameter<std::string>("type");
             std::string productInstanceName = pset.getParameter<std::string>("fromProductInstance");
             std::string instanceAlias = pset.getParameter<std::string>("toProductInstance");
-            if (productInstanceName == star) {
+
+            if (friendlyClassName == star) {
+              bool match = false;
+              for (auto it = moduleLabelToBranches.lower_bound(moduleLabel);
+                   it != moduleLabelToBranches.end() && it->first == moduleLabel;
+                   ++it) {
```

Comment:
At this point you know the moduleLabel is in the process. You could declare 'bool processHasLabel = false' before the loop and then set it true if you get here. Then you do not need the loop over the product registry below if there is no match.

Reply:
Thanks @wddgit, the logic can indeed simplified.